### PR TITLE
[CWS] do not copy process context for exec event, ptrace

### PR DESCRIPTION
**Pull request by safchain**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/probe/model.go [hunk 1]
```diff
@@ -209,20 +208,19 @@ func (ev *Event) ResolveContainerTags(e *model.ContainerContext) []string {
...
+// UnmarshalProcessCacheEntry unmarshal a Process
+func (ev *Event) UnmarshalProcessCacheEntry(data []byte) (int, error) {
 	// reset the process cache entry of the current event
 	entry := ev.resolvers.ProcessResolver.NewProcessCacheEntry()
-	entry.Pid = ev.ProcessContext.Pid
-	entry.Tid = ev.ProcessContext.Tid
+	entry.PIDContext = ev.PIDContext
 
 	n, err := entry.Process.UnmarshalBinary(data)
 	if err != nil {
```
Comment (lebauce) [inline pkg/security/probe/model.go:218]: We should release the process cache entry here

File: pkg/security/secl/model/model.go [hunk 2]
```diff
@@ -165,14 +164,18 @@ type Event struct {
...
-	Mount            MountEvent            `field:"-"`
-	Umount           UmountEvent           `field:"-"`
-	InvalidateDentry InvalidateDentryEvent `field:"-"`
-	ArgsEnvs         ArgsEnvsEvent         `field:"-"`
-	MountReleased    MountReleasedEvent    `field:"-"`
-	CgroupTracing    CgroupTracingEvent    `field:"-"`
-	NetDevice        NetDeviceEvent        `field:"-"`
-	VethPair         VethPairEvent         `field:"-"`
+	// internal usage
+	PIDContext        PIDContext            `field:"-"`
```
Comment (Gui774ume) [inline pkg/security/secl/model/model.go:168]: Would it make more sense to do something like this:

```go
type Event struct {
	ID           string    `field:"-"`
	Type         uint64    `field:"-"`
	TimestampRaw uint64    `field:"-"`
	Timestamp    time.Time `field:"-"` // Timestamp of the event

	    // Context shared with all events 
        ProcessCacheEntry *ProcessCacheEntry `field:"-"`
        PIDContext                PIDContext                `field:"-"`
        SpanContext             SpanContext              `field:"-"`
	ProcessContext        *ProcessContext       `field:"process" event:"*"`
	ContainerContext     ContainerContext      `field:"container"`
	NetworkContext       NetworkContext        `field:"network"`
	
        // FIM events
	Chmod       ChmodEvent    `field:"chmod" event:"chmod"`             // [7.27] [File] A file’s permissions were changed
	Chown       ChownEvent    `field:"chown" event:"chown"`             // [7.27] [File] A file’s owner was changed
	Open        OpenEvent     `field:"open" event:"open"`               // [7.27] [File] A file was opened
	Mkdir       MkdirEvent    `field:"mkdir" event:"mkdir"`             // [7.27] [File] A directory was created
	Rmdir       RmdirEvent    `field:"rmdir" event:"rmdir"`             // [7.27] [File] A directory was removed
	Rename      RenameEvent   `field:"rename" event:"rename"`           // [7.27] [File] A file/directory was renamed
	Unlink      UnlinkEvent   `field:"unlink" event:"unlink"`           // [7.27] [File] A file was deleted
	Utimes      UtimesEvent   `field:"utimes" event:"utimes"`           // [7.27] [File] Change file access/modification times
	Link        LinkEvent     `field:"link" event:"link"`               // [7.27] [File] Create a new name/alias for a file
	SetXAttr    SetXAttrEvent `field:"setxattr" event:"setxattr"`       // [7.27] [File] Set exteneded attributes
	RemoveXAttr SetXAttrEvent `field:"removexattr" event:"removexattr"` // [7.27] [File] Remove extended attributes
	Splice      SpliceEvent   `field:"splice" event:"splice"`           // [7.36] [File] A splice command was executed
	
        // process events
	Exec   ExecEvent   `field:"exec" event:"exec"`     // [7.27] [Process] A process was executed or forked
	SetUID SetuidEvent `field:"setuid" event:"setuid"` // [7.27] [Process] A process changed its effective uid
	SetGID SetgidEvent `field:"setgid" event:"setgid"` // [7.27] [Process] A process changed its effective gid
	Capset CapsetEvent `field:"capset" event:"capset"` // [7.27] [Process] A process changed its capacity set
	Signal SignalEvent `field:"signal" event:"signal"` // [7.35] [Process] A signal was sent
	
	    // kernel events
	SELinux      SELinuxEvent      `field:"selinux" event:"selinux"`             // [7.30] [Kernel] An SELinux operation was run
	BPF          BPFEvent          `field:"bpf" event:"bpf"`                     // [7.33] [Kernel] A BPF command was executed
	PTrace       PTraceEvent       `field:"ptrace" event:"ptrace"`               // [7.35] [Kernel] A ptrace command was executed
	MMap         MMapEvent         `field:"mmap" event:"mmap"`                   // [7.35] [Kernel] A mmap command was executed
	MProtect     MProtectEvent     `field:"mprotect" event:"mprotect"`           // [7.35] [Kernel] A mprotect command was executed
	LoadModule   LoadModuleEvent   `field:"load_module" event:"load_module"`     // [7.35] [Kernel] A new kernel module was loaded
	UnloadModule UnloadModuleEvent `field:"unload_module" event:"unload_module"` // [7.35] [Kernel] A kernel module was deleted
	DNS          DNSEvent          `field:"dns" event:"dns"`                     // [7.36] [Network] A DNS request was sent
	
	// internal usage events
	Mount             MountEvent            `field:"-"`
	Umount            UmountEvent           `field:"-"`
	InvalidateDentry  InvalidateDentryEvent `field:"-"`
	ArgsEnvs          ArgsEnvsEvent         `field:"-"`
	MountReleased     MountReleasedEvent    `field:"-"`
	CgroupTracing     CgroupTracingEvent    `field:"-"`
	NetDevice         NetDeviceEvent        `field:"-"`
	VethPair          VethPairEvent         `field:"-"`
}
```

File: pkg/security/probe/serializers.go [hunk 3]
```diff
@@ -499,7 +499,7 @@ func newProcessContextSerializer(pc *model.ProcessContext, e *Event, r *Resolver
 	if e == nil {
 		// custom events create an empty event
 		e = NewEvent(r, nil, nil)
-		e.ProcessContext = *pc
+		e.ProcessContext = pc
```
[the remark this answers is not in the document]
Comment (safchain) [reply pkg/security/probe/serializers.go:502]: We serialize before sending so I think we're good

File: pkg/security/probe/model.go (hunk 1 above)
Comment (safchain) [reply pkg/security/probe/model.go:218]: It is not Retained yet so no need to release it

Comment (safchain) [reply pkg/security/probe/model.go:218]: It is not Retained yet so no need to release it

File: pkg/security/probe/model.go [hunk 4]
```diff
@@ -470,25 +468,27 @@ func (ev *Event) ResolveEventTimestamp() time.Time {
...
-			// once process context without inode/mountid bug will be fixed
-			ev.processCacheEntry.FileEvent.SetPathnameStr("")
-			ev.processCacheEntry.FileEvent.SetBasenameStr("")
+	if ev.ProcessCacheEntry == nil {
+		ev.ProcessCacheEntry = ev.resolvers.ProcessResolver.Resolve(ev.PIDContext.Pid, ev.PIDContext.Tid)
+	}
+
+	if ev.ProcessCacheEntry == nil {
+		// keep the original PIDContext
+		ev.ProcessCacheEntry = &model.ProcessCacheEntry{
```
[the remark this answers is not in the document]
Comment (safchain) [reply pkg/security/probe/model.go:479]: Done

**Pull request merged**